File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for(int i=0; i<moves.length; i++) {
            int j=moves[i]-1;
            for (int k=0;k<board.length; k++){
                if(board[k][j] != 0) {
                    stack.push(board[k][j]);
                    board[k][j] = 0;
                    int stack_size = stack.size();
                    if(stack_size > 1 && stack.peek() == stack.get(stack_size-2)){
                        stack.pop();
                        stack.pop();
                        answer += 2;
                    }
                    break;
                    
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    stack = []
    cnt = 0
    
    for i in moves: # moves에서 하나 씩 값을 빼고
        for j in range(len(board)): # 위에서 아래로 돌면서
            if board[j][i-1] != 0: # 0이 아닐 때
                stack.append(board[j][i-1]) # 해당 값을 스택에 추가
                board[j][i-1] = 0 # 0으로 변경
                
                if len(stack) > 1: # 스택에 값이 2개 이상일 때
                    if stack[-1] == stack[-2]:
                        stack.pop()
                        stack.pop()
                        cnt += 2
                break
    return cnt